gap_linux.go

```diff
@@ -59,10 +60,20 @@ func (a *Advertisement) Start() error {
 	if a.advertisement != nil {
 		panic("todo: start advertisement a second time")
 	}
-	_, err := api.ExposeAdvertisement(a.adapter.id, a.properties, uint32(a.properties.Timeout))
+	cancel, err := api.ExposeAdvertisement(a.adapter.id, a.properties, uint32(a.properties.Timeout))
 	if err != nil {
 		return err
 	}
+	a.cancel = cancel
+	return nil
+}
+
+// Stop advertisement.
+func (a *Advertisement) Stop() error {
+	if a.cancel == nil {
+		panic("stop advertisement that was not started")
```

Comment:
It would be better to return an error here rather than `panic()`, please.

Reply:
@ysoldak see comment here, please.

Reply:
rebased and converted to error, @deadprogram